xbmc/utils/ActorProtocol.h

```diff
@@ -54,10 +55,18 @@ class Message
   Message() {isSync = false; data = NULL; event = NULL; replyMessage = NULL;};
 };
 
+class MessageHandle : public ::KODI::UTILS::CScopeGuard<Message*, nullptr, void(Message*)>
```

Comment:
In UML you define ports and connect them. The sender has a port and the receiver has a port. You are using it with just a single port